python/dgl/contrib/graph_partition_book.py

```diff
@@ -0,0 +1,254 @@
+            assert len(part) == self._num_partitions
+            start = 0
+            for offset in count:
+                part_nids = sorted_nid[start:start+offset]
+                start += offset
+                self._partid2nids.append(part_nids)
+
+        return self._partid2nids[partid]
+
+
+    def partid2eids(self, partid):
+        """From partition id to edge IDs
+
+        Parameters
+        ----------
+        partid : int
+            partition id
+
+        Returns
+        -------
+        tensor
+            edge IDs
+        """
+        if len(self._partid2eids) == 0:
+            sorted_eid = F.tensor(np.argsort(F.asnumpy(self._eid2partid)))
+            part, count = np.unique(F.asnumpy(self._eid2partid), return_counts=True)
+            assert len(part) == self._num_partitions
+            start = 0
+            for offset in count:
+                part_eids = sorted_eid[start:start+offset]
+                start += offset
+                self._partid2eids.append(part_eids)
+
+        return self._partid2eids[partid]
+
+    
+    def nid2localnid(self, nids, partid):
+        """Get local node IDs within the given partition.
+
+        Parameters
+        ----------
+        nids : tensor
+            global node IDs
+        partid : int
+            partition ID
+
+        Returns
+        -------
+        tensor
+             local node IDs
+        """
+        if len(self._nidg2l) == 0:
+            for partid in range(self._num_partitions):
+                global_id = self._part_graphs[partid].ndata[NID]
+                max_global_id = F.asnumpy(global_id).amax()
+                g2l = F.zeros((max_id+1), F.int64, F.cpu())
+                g2l[global_id] = F.arange(0, len(global_id))
+                self._nidg2l.append(g2l)
+
+        return self._nidg2l[partid]
+        
+
+    def eid2localeid(self, eids, partid):
+        """Get the local edge ids within the given partition.
+
+        Parameters
+        ----------
+        eids : tensor
+            global edge ids
+        partid : int
+            partition ID
+
+        Returns
+        -------
+        tensor
+             local edge ids
+        """
+        if len(self._eidg2l) == 0:
+            for part_id in range(self._num_partitions):
+                global_id = self._part_graphs[partid].edata[EID]
+                max_global_id = F.asnumpy(global_id).amax()
+                g2l = F.zeros((max_id+1), F.int64, F.cpu())
+                g2l[global_id] = F.arange(0, len(global_id))
+                self._eidg2l.append(g2l)
+
+        return self._eidg2l[partid]
+
+
+    def get_partition(self, partid):
+        """Get the graph of one partition.
+        
+        Parameters
+        ----------
+        partid : int
+            Partition ID.
+            
+        Returns
+        -------
+        DGLGraph
+            The graph of the partition.
+        """
+        return self._part_graphs[partid]
```

Comment:
Is it necessary to read remote partitions for now? I don't find any functions/models need to use it at this point. If we want to implement fetching remote partitions, it's related to distributed sampler which transmits a graph structure to remote machines. If we really want to read remote partitions, I would suggest implementing it after we have distributed sampler and underlying communicator ready. @jermainewang